File: 02_Lenguajes-de-programación/10_Go/07_estructura.go
package main

import ("fmt")

func main() {

	// Instanciado una estrcutura:
	myStruct := MyStruct{ "Diego", 23 }
	fmt.Println(myStruct)

}

// Estructura (Struct). Es similar al concepto de 'Clases'.
type MyStruct struct {
	name string
	age int
}

File: 02_Lenguajes-de-programación/09_Rust/07_estructura.rs
// Estructuras. Funciona como clases

struct MyStruct {
    name : String,
    age: i32,
}

impl MyStruct {
    fn new(name : &str, age: i32,) -> MyStruct{
        MyStruct {
            name :String::from(name),
            age
            }
    }
}

fn main() {

    // Instaciando una estructura
    let my_struct = MyStruct:: new("Diego", 23);
    println!("Mi nombre es {} y tengo {} años.", my_struct.name, my_struct.age);

}
